File: 卡码/代码随想录/JS/二叉树/02-二叉树属性/LeetCode_0111.js
//111. 二叉树的最小深度
import TreeNode from "../TreeNode.js";

const minDepth = function(root) {
  if(!root) return 0;
  if(root.left&&!root.right) return 1+minDepth(root.left);
  if(!root.left&&root.right) return 1+minDepth(root.right);
  return 1+Math.min(minDepth(root.left),minDepth(root.right));
};

File: 卡码/代码随想录/C++/二叉树/02-二叉树属性/LeetCode_0111.cpp
//111. 二叉树的最小深度
#include <iostream>
#include "../TreeNode.h"
using namespace std;

class Solution {
public:
  int minDepth(TreeNode* root) {
		if(root==nullptr) return 0;
		if(root->left!=nullptr&&root->right==nullptr) return 1+minDepth(root->left);
		if(root->left==nullptr&&root->right!=nullptr) return 1+minDepth(root->right);
		return 1+min(minDepth(root->left),minDepth(root->right));
  }
};

int main(){
  system("pause");
  return 0;
}
